src/router.ts

```diff
@@ -17,27 +18,41 @@ export function createRouter<T extends RadixNodeData = RadixNodeData>(
     staticRoutesMap: {},
   };
 
-  const normalizeTrailingSlash = (p) =>
+  const normalizeTrailingSlash = (p: string) =>
     options.strictTrailingSlash ? p : p.replace(/\/$/, "") || "/";
 
   if (options.routes) {
     for (const path in options.routes) {
-      insert(ctx, normalizeTrailingSlash(path), options.routes[path]);
+      Array.isArray(options.routes[path]) &&
+      options.routes[path].length === 2 &&
+      HTTPMethods.includes(options.routes[path][0])
+        ? insert(
+            ctx,
+            normalizeTrailingSlash(path),
+            options.routes[path][1],
+            options.routes[path][0]
+          )
+        : insert(ctx, normalizeTrailingSlash(path), options.routes[path]);
     }
   }
 
   return {
     ctx,
-    // @ts-ignore
-    lookup: (path: string) => lookup(ctx, normalizeTrailingSlash(path)),
-    insert: (path: string, data: any) =>
-      insert(ctx, normalizeTrailingSlash(path), data),
+    // @ts-expect-error - types are not matching
+    lookup: (path: string, method?: HTTPMethod) =>
```

Comment:
What is not clear in **acceptance criteria** that you decided to implement like this?

<img width="911" alt="image" src="https://github.com/unjs/radix3/assets/5158436/17bb0f81-9b0b-44a3-a51f-c66579c5f869">

Reply:
Changed to `{ method }`